### DISCUSSION TITLE
Arranging Widgets in a Grid (macOS style application launcher)

### DISCUSSION THREAD

**Discussion by WillPower3309** [Aylur/ags#429] [Q&A] [answered]
Hello! I'm curious how I could arrange widgets in a Grid. I have an application launcher I've been working on that aims to reproduce a macOS style app launcher. It's simple to get the icons to arrange in a vertical or horizontal list, but it doesn't seem like AGS has any sort of grid layout. Before I dive into trying to figure out a way to nest vertical and horizontal lists to achieve this, I wanted to check if anyone had a better solution, or has solved this problem before.

**Comment by kotontrion** [comment, marked as answer]
There are two ways to solve this.

You can use every gtk3 widget in ags, so you can just use the gtk grid widget. However you have to manually specify the coordinates of every widget in grid when adding them.

I recommend using Widget.FlowBox, which I also use in my applauncher. A flowbox is basically like a box, but it will arrange its children in multiple rows automatically when required.

**Reply by WillPower3309** [reply]
Thanks for the insight! I'll take a look at the flowbox
